sencha-workspace/SlateTasksStudent/app/view/TaskTree.js

```diff
@@ -186,33 +186,44 @@ Ext.define('SlateTasksStudent.view.TaskTree', {
             store = me.getStore(),
             items = [],
 
-            rootTasks = store.queryBy(function(studentTask) {
-                return !studentTask.get('Task').ParentTaskID;
+            topLevelTaskIds = [],
+            topLevelTasks = store.queryBy(function(studentTask) {
+                if (!studentTask.get('Task').ParentTaskID) {
+                    topLevelTaskIds.push(studentTask.get('TaskID'));
```

Comment:
@nbey I don't know if it's a good idea to be "doing work" during a query loop, these functions should only return true/false. Since returning true corresponds with this action, I think you could just do something like `topLevelTaskIds = topLevelTasks.collect('TaskID')` on the next line